Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1244,6 +1244,13 @@ private static void CleanupCheckers(string requestId)
             verificationResult.Errors = null;
             verificationResult.Outcome = VCGen.Outcome.Inconclusive;
           }
+          catch(AggregateException e)
```

Comment:
Can we catch something more specific? I wouldn't want to swallow any bugs here but catching a bad solver argument seems good.

Reply:
There's nothing below `AggregateException` in the class hierarchy, AFAIK, though we could pull out the wrapped exceptions and behave differently depending on what they are. This code will always lead to a message and an inconclusive result, so I wouldn't expect it to hide bugs. It could be that, after some experience with it, we realize that it would be best to handle different exceptions differently, but at the moment I'm not sure what the different behavior would be.

At the moment, this does catch bad solver arguments and paths, but also things like a broken pipe if a solver crashes, or exceptions thrown by internal code that relates to solver interaction (like failing to parse a solver result).

Reply:
Inspecting the internals of `AggregateException` and rethrowing if it doesn't contain something we want to handle, seems fine to me.

Reply:
That sounds good to me. I'll update it to look inside the `AggregateException`.

Reply:
One possibility is to _only_ handle `IOException` here. Solver configuration issues result in a `ProverException` which is then handled higher up, meaning the output expected by the current solver configuration error tests doesn't change.

Reply:
The latest commit does just this, and I think I like the way it behaves now (as detailed in the response below).